src/ws.c

```diff
@@ -297,22 +325,29 @@ int ws_socket(struct ws_events *evs, int port)
 	printf("Waiting for incoming connections...\n");
 
 	len = sizeof(struct sockaddr_in);
-	
+	memset(client_socks, 0, sizeof(client_socks));
+	/* Initialize global mutex. */
+	pthread_mutex_init(&mutex, NULL);
 	/* Accept connections. */
 	while (1)
 	{
 		/* Accept. */
 		new_sock = accept(sock, (struct sockaddr *)&client, (socklen_t*)&len);
 		if (new_sock < 0)
 		{
-			perror("Error on accepting conections..");
+			perror("Error on accepting connections..");
 			exit(-1);
 		}
-
+		/* Adds client socket to socks list. */
+	    pthread_mutex_lock(&mutex);
+	    client_socks[client_count++] = new_sock;
```

Comment:
- The 'free slot' (in client_socks) is not guaranteed to be continuous, since already connected clients can disconnect and release slots in the 'middle' of the vector.

  What I suggest to you, is to do something similar to what was done in `ws_receiveframe` where you remove the client from the socket list, something like:
```c
pthread_mutex_lock(&mutex);

for (unsigned int i = 0; i < MAX_CLIENTS; i++)
	if (client_socks[i] == -1)
		client_socks[i] = new_sock, i = MAX_CLIENTS;

pthread_mutex_unlock(&mutex);
```
*Keep in mind that changing the default client_socks value from 0 to -1 also needs to change the line 328: instead of using memset, iterate over the vector setting the elements to -1.

- The variable `client_count` is not used anywhere, please consider remove it.